app/coffee/RoomManager.coffee

```diff
@@ -66,6 +68,12 @@ module.exports = RoomManager =
             callback()
 
     leaveEntity: (client, entity, id) ->
+        # Ignore any requests to leave when the client is not actually in the
+        # room. This can happen if the client sends spurious leaveDoc requests
+        # for old docs after a reconnection.
+        if !@_clientAlreadyInRoom(client, id)
```

Comment:
Coming from `leaveProjectAndDocs` this is a static `FALSE`.

From your PR description I'd say this check should rather go into the `leaveDoc` method.

Reply:
> Coming from leaveProjectAndDocs this is a static FALSE.

True, but it seems safer to do the check that the client is in the room at the point where the `leave` is called.